cloudant-sync-datastore-core/src/test/java/com/cloudant/sync/internal/replication/PullStrategyMockTest.java

```diff
@@ -231,6 +232,97 @@ public Object answer(InvocationOnMock invocation) throws Throwable {
+    }
+
+    @Test
+    public void testDoNotSetCheckpointWhenNotModified() throws Exception {
+        CouchDB mockRemoteDb = mock(CouchDB.class);
+        when(mockRemoteDb.changes((PullFilter) null, "10-d9e5b0147af143e5b6d1979378ad957b", 1000)).then(new Answer<Object>() {
+            @Override
+            public Object answer(InvocationOnMock invocation) throws Throwable {
+                FileReader fr = new FileReader(TestUtils.loadFixture
+                        ("fixture/empty_changes.json"));
+                return JSONUtils.fromJson(fr, ChangesResult.class);
+            }
+        });
+        when(mockRemoteDb.exists()).thenReturn(true);
+
+        DatastoreWrapper mockLocalDb = mock(DatastoreWrapper.class);
+        when(mockLocalDb.getCheckpoint(any(String.class))).thenReturn("10" +
+                "-d9e5b0147af143e5b6d1979378ad957b");
+
+        StrategyListener mockListener = mock(StrategyListener.class);
+        PullStrategy pullStrategy = super.getPullStrategy();
+        pullStrategy.sourceDb = mockRemoteDb;
+        pullStrategy.targetDb = mockLocalDb;
+        pullStrategy.getEventBus().register(mockListener);
+        pullStrategy.run();
+
+        //make sure the correct events were fired
+        verify(mockListener).complete(any(ReplicationStrategyCompleted.class));
+        verify(mockListener, never()).error(any(ReplicationStrategyErrored.class));
+        verify(mockLocalDb, never()).putCheckpoint(anyString(), any());
+    }
+
+    @Test
+    public void testSetCheckpointWhenModified() throws Exception {
+        CouchDB mockRemoteDb = mock(CouchDB.class);
+        when(mockRemoteDb.changes((PullFilter) null, "9-d9e5b0147af143e5b6d1979378ad957b", 1000)).then(new Answer<Object>() {
+            @Override
+            public Object answer(InvocationOnMock invocation) throws Throwable {
+                FileReader fr = new FileReader(TestUtils.loadFixture
+                        ("fixture/empty_changes.json"));
+                return JSONUtils.fromJson(fr, ChangesResult.class);
+            }
+        });
+        when(mockRemoteDb.exists()).thenReturn(true);
+
+        DatastoreWrapper mockLocalDb = mock(DatastoreWrapper.class);
+        when(mockLocalDb.getCheckpoint(any(String.class))).thenReturn("9" +
+                "-d9e5b0147af143e5b6d1979378ad957b");
+
+        StrategyListener mockListener = mock(StrategyListener.class);
+        PullStrategy pullStrategy = super.getPullStrategy();
+        pullStrategy.sourceDb = mockRemoteDb;
+        pullStrategy.targetDb = mockLocalDb;
+        pullStrategy.getEventBus().register(mockListener);
+        pullStrategy.run();
+
+        //make sure the correct events were fired
+        verify(mockListener).complete(any(ReplicationStrategyCompleted.class));
+        verify(mockListener, never()).error(any(ReplicationStrategyErrored.class));
+        verify(mockLocalDb).putCheckpoint(anyString(), eq("10-d9e5b0147af143e5b6d1979378ad957b"));
```

Comment:
As above, `try`/`catch` with an `assumeNoException` to skip this when the class can't be proxied.